simplemma/dictionary_factory.py

```diff
@@ -33,20 +33,26 @@ def _load_dictionary_from_disk(langcode: str) -> Dict[str, str]:
 
 
 class DictionaryFactory:
+    __slots__ = ["_data", "_load_dictionary_from_disk"]
+
     def __init__(self, cache_max_size: Optional[int] = 8):
         self._data: Dict[str, Dict[str, str]] = {}
         self._load_dictionary_from_disk = lru_cache(maxsize=cache_max_size)(
             _load_dictionary_from_disk
         )
 
     def get_dictionaries(
-        self, langs: Optional[Union[str, Tuple[str]]]
+        self,
+        langs: Optional[Union[str, Tuple[str, ...]]] = None,
     ) -> Dict[str, Dict[str, str]]:
+        # check validity of input
         langs = _control_lang(langs)
 
-        if self._data and tuple(sorted(self._data.keys())) == sorted(langs):
+        # check if the languages already loaded can be re-used
+        if sorted(self._data.keys()) == sorted(langs):
             return self._data
 
+        # (re)load data
```

Comment:
Same thing here.

If you want to make this more clear, we could move this block to an internal method called `load_dictionaries` and get rid of the comment

Reply:
OK, let's remove the comment